README.md

```diff
@@ -25,122 +25,95 @@ Apollo-android features:
-2) Add your `schema.json` to the directory at `src/main/graphql/com/example/schema.json`. If you don't have a `schema.json` file yet, read the section about [downloading a schema file](#downloading-a-schemajson-file). 
-3) Put your GraphQL queries in a `.graphql` files. For an example: `src/main/graphql/com/example/feed.graphql`: 
+Apollo-Android requires an introspection schema. You can get a schema.json file by running an introspection query on your endpoint. The Apollo Gradle plugin exposes a `downloadApolloSchema` task to help with this. You can download a schema by specifying your endpoint and the location where you want the schema to be downloaded:
 
+```bash
+./gradlew downloadApolloSchema --endpoint=https://your.graphql.endpoint --schema=app/src/main/graphql/com/example/schema.json
 ```
-query FeedQuery($type: FeedType!, $limit: Int!) {
-  feed(type: $type, limit: $limit) {
-    comments {
-      ...FeedCommentFragment
-    }
-    repository {
-      ...RepositoryFragment
-    }
-    postedBy {
-      login
-    }
-  }
-}
 
-fragment RepositoryFragment on Repository {
-  name
-  full_name
-  owner {
-    login
-  }
-}
+If your endpoint requires authentication, you can pass custom HTTP headers:
+
+```
+./gradlew downloadApolloSchema --endpoint="https://your.graphql.endpoint" --schema="app/src/main/graphql/com/example" --header="Authorization: Bearer $TOKEN"
+```
 
-fragment FeedCommentFragment on Comment {
-  id
-  postedBy {
-    login
+## Generating models from your queries
+
+1) Create a directory for your GraphQL files like you would do for Java/Kotlin: `src/main/graphql/com/example/`. Apollo-Android will generate models in the `com.apollographql.apollo.sample` package.
+2) Add your `schema.json` to the directory at `src/main/graphql/com/example/schema.json`. If you don't have a `schema.json` file yet, read the section about [downloading a schema file](#downloading-a-schemajson-file). 
+3) Put your GraphQL queries in a `.graphql` files. For an example: `src/main/graphql/com/example/LaunchDetails.graphql`: 
+
+```graphql
+query LaunchDetails($id:ID!) {
+  launch(id: $id) {
+    id
+    site
+    mission {
+      name
+      missionPatch(size:LARGE)
+    }
   }
-  content
 }
 ```
 
 4) Decide if you want to generate Kotlin or Java models:
 
 ```groovy
-// build.gradle or build.gradle.kts
+// app/build.gradle or app/build.gradle.kts
 apollo {
   generateKotlinModels.set(true) // or false for Java models
 }
 ```
 
-5) Execute `./gradlew generateApolloSources` to generate the models from your queries. This will create a generated `FeedQuery` Java or Kotlin source file for your query.
+5) Execute `./gradlew generateApolloSources` to generate the models from your queries. This will create a generated `LaunchDetailsQuery` Java or Kotlin source file for your query.
```

Comment:
This always confused me. Explicit running of this task is not necessary. Building/assembling also works.

Reply:
Sounds good, I'll remove this.